controller/controlplane/manager_options.go

```diff
@@ -93,3 +96,109 @@ func WithMetricsServerOff() managercfg.Opt {
 		c.MetricsAddr = "0" // 0 disables metrics server
 	}
 }
+
+// WithFeatureGates sets the feature gates for the manager.
+func WithFeatureGates(logger logr.Logger, featureGates []gwtypes.ControlPlaneFeatureGate) managercfg.Opt {
+	return func(c *managercfg.Config) {
+		fgs := managercfg.FeatureGates{}
+		defaults := managercfg.GetFeatureGatesDefaults()
+		for _, feature := range featureGates {
+			if _, ok := defaults[feature.Name]; !ok {
+				log.Error(logger, errors.New("unknown feature gate"), "unknown feature gate",
+					"feature", feature.Name, "state", feature.State,
+				)
+				continue
+			}
+
+			v := false
+			if feature.State == gwtypes.FeatureGateStateEnabled {
+				v = true
+			}
+			// This should never happen as it should be enforced at the CRD level
+			// but we handle it gracefully here and log an error.
+			if _, ok := fgs[feature.Name]; ok {
+				log.Error(logger, errors.New("feature gate already set"), "feature gate already set",
+					"feature", feature.Name, "state", feature.State,
+				)
+				continue
+			}
+			fgs[feature.Name] = v
+		}
+
+		for k, v := range defaults {
+			// Ensure that we don't override the defaults with empty values
+			if _, ok := fgs[k]; !ok {
+				fgs[k] = v
+			}
+		}
+		c.FeatureGates = fgs
+	}
+}
+
+// WithControllers sets the controllers for the manager.
+func WithControllers(logger logr.Logger, controllers []gwtypes.ControlPlaneController) managercfg.Opt {
+	logDeprecated := func(logger logr.Logger, enabled bool, controllerName string) {
+		if enabled {
+			log.Info(logger, "chosen controller is deprecated", "controller", controllerName)
+		}
+	}
+	setOpt := func(b *bool, state gwtypes.ControllerState) {
+		if b == nil {
+			return
+		}
+		*b = state == gwtypes.ControlPlaneControllerStateEnabled
```

Comment:
```suggestion
		*b = (state == gwtypes.ControlPlaneControllerStateEnabled)
```
Could we use brackets to make it clearer?

Reply:
Sure. Applied the suggestion.